src/model/AirflowNetworkSpecifiedFlowRate.cpp

```diff
@@ -0,0 +1,150 @@
+      : AirflowNetworkComponent_Impl(other, model, keepHandle) {
+      OS_ASSERT(other.iddObject().type() == AirflowNetworkSpecifiedFlowRate::iddObjectType());
+    }
+
+    AirflowNetworkSpecifiedFlowRate_Impl::AirflowNetworkSpecifiedFlowRate_Impl(const AirflowNetworkSpecifiedFlowRate_Impl& other, Model_Impl* model,
+                                                                               bool keepHandle)
+      : AirflowNetworkComponent_Impl(other, model, keepHandle) {}
+
+    const std::vector<std::string>& AirflowNetworkSpecifiedFlowRate_Impl::outputVariableNames() const {
+      static const std::vector<std::string> result;
+      return result;
+    }
+
+    IddObjectType AirflowNetworkSpecifiedFlowRate_Impl::iddObjectType() const {
+      return AirflowNetworkSpecifiedFlowRate::iddObjectType();
+    }
+
+    double AirflowNetworkSpecifiedFlowRate_Impl::airFlowValue() const {
+      boost::optional<double> value = getDouble(OS_AirflowNetworkSpecifiedFlowRateFields::AirFlowValue, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    std::string AirflowNetworkSpecifiedFlowRate_Impl::airFlowUnits() const {
+      boost::optional<std::string> value = getString(OS_AirflowNetworkSpecifiedFlowRateFields::AirFlowUnits, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    bool AirflowNetworkSpecifiedFlowRate_Impl::isAirFlowUnitsDefaulted() const {
+      return isEmpty(OS_AirflowNetworkSpecifiedFlowRateFields::AirFlowUnits);
+    }
+
+    bool AirflowNetworkSpecifiedFlowRate_Impl::setAirFlowValue(double airFlowValue) {
+      bool result = setDouble(OS_AirflowNetworkSpecifiedFlowRateFields::AirFlowValue, airFlowValue);
+      return result;
+    }
+
+    bool AirflowNetworkSpecifiedFlowRate_Impl::setAirFlowUnits(const std::string& airFlowUnits) {
+      bool result = setString(OS_AirflowNetworkSpecifiedFlowRateFields::AirFlowUnits, airFlowUnits);
+      return result;
+    }
+
+    void AirflowNetworkSpecifiedFlowRate_Impl::resetAirFlowUnits() {
+      bool result = setString(OS_AirflowNetworkSpecifiedFlowRateFields::AirFlowUnits, "");
+      OS_ASSERT(result);
+    }
+
+  }  // namespace detail
+
+  AirflowNetworkSpecifiedFlowRate::AirflowNetworkSpecifiedFlowRate(const Model& model, double airFlowValue)
+    : AirflowNetworkComponent(AirflowNetworkSpecifiedFlowRate::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::AirflowNetworkSpecifiedFlowRate_Impl>());
+
+    bool ok = setAirFlowValue(airFlowValue);
+    OS_ASSERT(ok);
+  }
+
+  IddObjectType AirflowNetworkSpecifiedFlowRate::iddObjectType() {
+    return IddObjectType(IddObjectType::OS_AirflowNetworkSpecifiedFlowRate);
+  }
+
+  std::vector<std::string> AirflowNetworkSpecifiedFlowRate::validAirFlowUnitsValues() {
+    return getIddKeyNames(IddFactory::instance().getObject(iddObjectType()).get(), OS_AirflowNetworkSpecifiedFlowRateFields::AirFlowUnits);
+  }
```

Comment:
validXXXValues() for a choice field, good! I kinda like adding also `xxxValues()` as an alternate name (because the API is inconsistent... so adding both)